tools/matc/src/matc/ParametersProcessor.cpp

```diff
@@ -1248,14 +1407,54 @@ bool ParametersProcessor::process(MaterialBuilder& builder, const JsonishObject&
             return false;
         }
 
-        auto fPointer = mParameters[key].callback;
+        auto fPointer = mParameters[key].jsonCallback;
         bool ok = fPointer(builder, *field);
         if (!ok) {
-            std::cerr << "Error while processing material key:\"" << key << "\"" << std::endl;
+            std::cerr << "Error while processing material json, key:\"" << key << "\"" << std::endl;
             return false;
         }
     }
     return true;
 }
 
+bool ParametersProcessor::process(filamat::MaterialBuilder& builder, const std::string& key, const std::string& value) {
+    if (mParameters.find(key) == mParameters.end()) {
+        std::cerr << "Ignoring config entry (unknown key): \"" << key << "\"" << std::endl;
+        return false;
+    }
+
+    auto fPointer = mParameters[key].variantCallback;
+    if (!fPointer) {
+        std::cerr << "No variant callback available, key:\"" << key << "\"" << std::endl;
+        return false;
+    }
+
+    Variant var;
+    switch (mParameters.at(key).rootAssert) {
```

Comment:
I'm fine with adding default, but recently I heard we prefer not having default in favor of a warning in case a new enum is added later. Which one would be the norm for Filament?

Reply:
In that case, we can add the missing enums as a no-op so we still generate a warning if a new one is added.

Reply:
But compiler wouldn't trigger a warning with a default label unless we build it with -Wswitch-enum option. And I don't see that flag in Filament.